comp/collector/collector/collectorimpl/internal/middleware/check_wrapper.go

```diff
@@ -32,20 +36,29 @@ type CheckWrapper struct {
 }
 
 // NewCheckWrapper returns a wrapped check.
-func NewCheckWrapper(inner check.Check, senderManager sender.SenderManager) *CheckWrapper {
+func NewCheckWrapper(inner check.Check, senderManager sender.SenderManager, agentTelemetry option.Option[agenttelemetry.Component]) *CheckWrapper {
 	return &CheckWrapper{
-		inner:         inner,
-		senderManager: senderManager,
+		inner:          inner,
+		senderManager:  senderManager,
+		agentTelemetry: agentTelemetry,
 	}
 }
 
 // Run implements Check#Run
-func (c *CheckWrapper) Run() error {
+func (c *CheckWrapper) Run() (err error) {
```

Comment:
For my understanding. Why add the `err` part to the return value?

Reply:
That way we can easily attach the error to the current span